pkg/util/deployment/deployment.go

```diff
@@ -53,9 +55,12 @@ func GetOldReplicaSets(deployment extensions.Deployment, c clientset.Interface)
 		})
 }
 
+type rsListFunc func(string, api.ListOptions) ([]extensions.ReplicaSet, error)
```

Comment:
I put these special-casing interfaces because `getOldReplciaSets` and `getNewReplicaSet` are called both in deployment controller and in deployment util. In controller we use rsStore and podStore to `List`, but in deployment util we use clientset.Interface to `List`. They're of different types (`storeXxxNamespacer` vs. `XxxInterface`) so I'm not sure how to avoid special-casing interfaces.